grpc/server.go

```diff
@@ -0,0 +1,87 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/agglayer/aggkit/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+type ServerConfig struct {
+	// Host is the address to bind the gRPC server
+	Host string `mapstructure:"Host"`
+	// Port is the port to bind the gRPC server
+	Port int `mapstructure:"Port"`
+	// EnableReflection indicates whether gRPC server reflection is enabled
+	EnableReflection bool `mapstructure:"EnableReflection"`
+}
+
+// Server encapsulates a gRPC server instance, its network listener, and the address it listens on.
+// It provides the necessary fields to manage the lifecycle and configuration of a gRPC server.
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	addr       string
+
+	cfg ServerConfig
+}
+
+// NewServer creates and returns a new gRPC Server instance listening on the specified address.
+// It accepts an address string, a boolean to enable server reflection, and optional gRPC server options.
+// If enableReflection is true, the server will register the reflection service for introspection.
+// Returns a pointer to the Server and an error if the listener cannot be created.
+func NewServer(cfg ServerConfig, opts ...grpc.ServerOption) (*Server, error) {
+	addr := fmt.Sprintf("%s:%d", cfg.Host, cfg.Port)
+
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	server := grpc.NewServer(opts...)
+
+	if cfg.EnableReflection {
+		reflection.Register(server) // Register reflection service on gRPC server
+	}
```

Comment:
WDYT about enabling this only in development/debugging environments? I'm suggesting to rely on logger parameter that has production and development values to determine which setup we are running currently?

Reply:
I would like to keep it. It is more easier and more flexible like this.